File: public/docs/_examples/attribute-directives/ts/src/app/highlight.directive.2.ts
// #docregion
import {Directive, ElementRef, Renderer, Input} from 'angular2/core';

@Directive({
  selector: '[my-highlight]',
  // #docregion host
  host: {
    '(mouseenter)': 'onMouseEnter()',
    '(mouseleave)': 'onMouseLeave()'
  }
  // #enddocregion host
})

export class Highlight {
  // #docregion ctor
  constructor(private el: ElementRef, private renderer: Renderer) {
  }
  // #enddocregion ctor

  // #docregion mouse-methods
  onMouseEnter() { this._highlight("yellow"); }
  onMouseLeave() { this._highlight(null); }

  private _highlight(color: string) {
    this.renderer.setElementStyle(this.el, 'background-color', color);
  }
  // #enddocregion mouse-methods

}
// #enddocregion

File: public/docs/_examples/attribute-directives/dart/lib/highlight_directive_2.dart
// #docregion
import 'package:angular2/core.dart';

@Directive(selector: '[myHighlight]')
class HighlightDirective {
  // #docregion ctor
  final dynamic _el;

  HighlightDirective(ElementRef elRef) : _el = elRef.nativeElement;
  // #enddocregion ctor

  // #docregion mouse-methods, host
  @HostListener('mouseenter')
  void onMouseEnter() {
    // #enddocregion host
    _highlight('yellow');
    // #docregion host
  }

  @HostListener('mouseleave')
  void onMouseLeave() {
    // #enddocregion host
    _highlight();
    // #docregion host
  }
  // #enddocregion host

  void _highlight([String color]) {
    if (_el != null) _el.style.backgroundColor = color;
  }
  // #enddocregion mouse-methods
}
// #enddocregion
